app/src/main/java/com/liato/bankdroid/BankdroidApplication.java

```diff
@@ -24,22 +22,16 @@
     @Override
     public void onCreate() {
         super.onCreate();
-        Fabric.with(this, new Crashlytics(), new Answers());
-        Crashlytics.setString(LOG_KEY_LOCALE, Locale.getDefault().toString());
-        logBanks();
-    }
 
-    private void logBanks() {
-        List<Bank> banks = BankFactory.banksFromDb(this, false);
-        StringBuilder bankStringBuilder = new StringBuilder();
-        if (banks != null && !banks.isEmpty()) {
-            for (Bank bank : banks) {
-                bankStringBuilder.append(bank.getName())
-                        .append(",");
-            }
-            Crashlytics.setString(LOG_KEY_BANKS,
-                    bankStringBuilder.substring(0, bankStringBuilder.length() - 1));
+        CrashlyticsCore core = new CrashlyticsCore.Builder()
+                .disabled(BuildConfig.DEBUG)
+                .build();
+        Fabric.with(this, new Crashlytics.Builder().core(core).build());
+
+        if (BuildConfig.DEBUG) {
+            Timber.plant(new Timber.DebugTree());
         }
+        Timber.plant(new CrashlyticsTree());
```

Comment:
The logic you want is rather "Crashlytics on devices, not-Crashlytics on the emulator, DEBUG-vs-RELEASE is not a factor".

Also, this needs to be done at all entry points to the program; the services as well. Check `AndroidManifest.xml` for inspiration, basically all `onCreate()` implementations should set up Crashlytics.

Feel free to copy code from / copy the entire files from here and relicense it for Bankdroid if you want:
https://github.com/walles/batterylogger/blob/master/src/main/java/com/gmail/walles/johan/batterylogger/LoggingUtils.java
https://github.com/walles/batterylogger/blob/master/src/main/java/com/gmail/walles/johan/batterylogger/EmulatorUtils.java